src/HonzaBotner.Discord.Services/Commands/Polls/PollCommand.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using DSharpPlus;
+using DSharpPlus.Entities;
+using HonzaBotner.Discord.Command;
+using Microsoft.Extensions.Logging;
+
+namespace HonzaBotner.Discord.Services.Commands.Polls
+{
+    public class PollCommand : BaseCommand
+    {
+        public const string ChatCommand = "poll";
+        protected override CommandPermission RequiredPermission => CommandPermission.Authorized;
+        protected override bool CanBotExecute => false;
+
+        public PollCommand(IPermissionHandler permissionHandler, ILogger logger)
+            : base(permissionHandler, logger)
+        {
+        }
+
+        protected override async Task<ChatCommendExecutedResult> ExecuteAsync(DiscordClient client, DiscordMessage message,
+            CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var parameters = ParseParameters(message.Content);
+                var poll = CreatePoll(client, message.Author, parameters);
+
+                await poll.Post(client, message.Channel);
+                await message.DeleteAsync();
+
+                return ChatCommendExecutedResult.Ok;
+            }
+            catch (ArgumentException ex)
+            {
+                await message.RespondAsync(ex.Message);
+                return ChatCommendExecutedResult.WrongSyntax;
+            }
+        }
+
+        protected IPoll CreatePoll(DiscordClient client, DiscordUser pollAuthor, List<string> parameters)
+        {
+            if (parameters.Count == 1)
+            {
+                return new YesNoPoll(pollAuthor.Username, pollAuthor.AvatarUrl, parameters[0]);
+            }
+
+            return new AbcPoll(pollAuthor.Username, pollAuthor.AvatarUrl, parameters[0], parameters.Skip(1).ToList());
+        }
+
+        protected List<string> ParseParameters(string message)
+        {
+            if (message.Count(c => c == '"') % 2 != 0)
+            {
+                throw new ArgumentException("Odd number of quotes");
+            }
+
+            const string quotedGroupPattern = @"""([^\""]*?)""";
```

Comment:
nit: Maybe add comment explaining this regex...

Reply:
Sure, it's kinda cryptic.